DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/DicomUtility.cs

```diff
@@ -0,0 +1,167 @@
+
+            if (groups["timeZone"].Success)
+            {
+                int sign = int.Parse(groups["sign"].Value + "1");
+                int timeZoneHour = int.Parse(groups["timeZoneHour"].Value) * sign;
+                int timeZoneMinute = int.Parse(groups["timeZoneMinute"].Value) * sign;
+                return new DateTimeObject()
+                {
+                    DateValue = new DateTimeOffset(year, month, day, hour, minute, second, millisecond, new TimeSpan(timeZoneHour, timeZoneMinute, 0)),
+                    HasTimeZone = true,
+                };
+            }
+            else
+            {
+                return new DateTimeObject()
+                {
+                    DateValue = new DateTimeOffset(year, month, day, hour, minute, second, millisecond, default),
+                    HasTimeZone = false,
+                };
+            }
+        }
+
+        public static string GenerateDicomDateString(DateTimeOffset date)
+        {
+            return date.ToString("yyyyMMdd", CultureInfo.InvariantCulture);
+        }
+
+        public static string GenerateDicomDateTimeString(DateTimeObject date)
+        {
+            EnsureArg.IsNotNull(date, nameof(date));
+
+            return date.HasTimeZone == true
+                ? date.DateValue.ToString("yyyyMMddHHmmss.ffffffzzz", CultureInfo.InvariantCulture).Replace(":", string.Empty)
+                : date.DateValue.ToString("yyyyMMddhhmmss.ffffff", CultureInfo.InvariantCulture);
+        }
+
+        public static AgeObject ParseAge(string age)
+        {
+            EnsureArg.IsNotNull(age, nameof(age));
+
+            foreach (var item in AgeTypeMapping)
+            {
+                if (new Regex(@"^\d{3}" + item.Key + "$").IsMatch(age))
+                {
+                    return new AgeObject(uint.Parse(age.Substring(0, AgeStringLength)), item.Value);
+                }
+            }
+
+            throw new DicomDataException("Invalid age string. The valid strings are nnnD, nnnW, nnnM, nnnY.");
+        }
+
+        public static string GenerateAgeString(AgeObject age)
+        {
+            EnsureArg.IsNotNull(age, nameof(age));
+
+            foreach (var item in AgeTypeMapping)
+            {
+                if (age.AgeType == item.Value)
+                {
+                    // Age string only support 3 charaters
+                    if (age.Value <= MaxAgeValue)
+                    {
+                        return age.Value.ToString().PadLeft(AgeStringLength, '0') + item.Key;
+                    }
+                    else if (age.AgeInYears() <= MaxAgeValue)
+                    {
+                        return age.AgeInYears().ToString().PadLeft(AgeStringLength, '0') + "Y";
+                    }
+
+                    throw new DicomDataException("Invalid age value for DICOM. The valid strings are nnnD, nnnW, nnnM, nnnY.");
```

Comment:
🟡 [Suggestion]
Better to revise the message.
Actually here is the case the age value > 999 can't be represented by an AgeString.
Current message is suitable for parsing error.
```suggestion
                    throw new DicomDataException($"Invalid age value [{age.Value}] for DICOM. It can't be represented by nnnD, nnnW, nnnM, nnnY.");
```